src/ZoneServer/Network/PacketHandler.cs

```diff
@@ -764,8 +764,8 @@ public void CZ_ITEM_USE(IZoneConnection conn, Packet packet)
 					return;
 				}
 
-				// Remove consumeable items on success
-				if (item.Data.Type == ItemType.Consume)
+				// Remove consumeable items on success, unless it's a Blue Orb and the Pet System is turned on
```

Comment:
While this is a simple way to accomplish not consuming certain types of items, I feel like hardcoding this sets a bad precedent. The next person with this problem might think this is the way to go and add their exception as well, then comes another and another and we quickly have quite the mess on our hands. Instead, we should think of a way to let the item usage function specify whether to skip the consumption of the item.

Reply:
I agree, though I didn't see any other way to achieve this with the current implementation except for possibly changing the item database, which I think is even worse.

Reply:
I've added a third item use result type that can be used to specify that a consumable should not be decremented.

Reply:
I've updated the code so that it uses it.  This does seem cleaner.